File: src/main/java/me/cozease/bookstorebackend/service/CartService.java
package me.cozease.bookstorebackend.service;

import me.cozease.bookstorebackend.entity.CartItem;
import org.springframework.transaction.annotation.Transactional;

import java.util.List;

/**
 * 购物车服务类
 */
@Transactional
public interface CartService {
    /**
     * 添加商品到购物车
     */
    CartItem addToCart(Long userId, Long bookId, Integer quantity);

    /**
     * 获取用户的购物车列表
     */
    List<CartItem> getCartItems(Long userId);

    /**
     * 更新购物车商品数量
     */
    CartItem updateCartItemQuantity(Long cartItemId, Integer quantity);

    /**
     * 从购物车中删除商品
     */
    void removeFromCart(Long cartItemId);

    /**
     * 清空用户的购物车
     */
    void clearCart(Long userId);
}

File: src/services/cartService.js
import request from './request';

export const CartService = {
    // 获取购物车列表
    getCartItems: async (userId) => {
        return request(`/cart/${userId}`);
    },

    // 添加商品到购物车
    addToCart: async (userId, bookId, quantity) => {
        return request(`/cart/add?userId=${userId}&bookId=${bookId}&quantity=${quantity}`, {
            method: 'POST'
        });
    },

    // 更新购物车商品数量
    updateCartItem: async (cartItemId, quantity) => {
        return request(`/cart/update/${cartItemId}?quantity=${quantity}`, {
            method: 'PUT'
        });
    },

    // 从购物车移除商品
    removeFromCart: async (cartItemId) => {
        return request(`/cart/${cartItemId}`, {
            method: 'DELETE',
        });
    },

    // 清空购物车
    clearCart: async (userId) => {
        return request(`/cart/clear/${userId}`, {
            method: 'DELETE',
        });
    },
};
